src/Flysystem/AssetAdapter.php

```diff
@@ -195,4 +195,36 @@ protected function renderTemplate($template)
 
         return $result;
     }
+
+    /**
+     * @param string $directory
+     * @param callable $filter
+     * @return \Generator Returns a generator for recursively iterating through a directory.
+     * This is intended to be less memory intensive than `listContents`
+     * as it yields per directory instead of returning the whole directory tree as one array.
+     * @see listContents
+     */
+    public function getFileGenerator(string $directory, $filter = null)
+    {
+        $files = $this->listContents($directory, false);
+        foreach ($files as $file) {
+            if (is_callable($filter) && !$filter($file)) {
+                continue;
+            }
+
+            yield $file;
+            if ($file['type'] == 'dir') {
+                yield from $this->getFileGenerator($file['path']);
```

Comment:
Didn't know about `yield from`. Looks pretty cool.

[But it doesn't work in PHP 5.6](https://www.php.net/manual/en/language.generators.syntax.php#control-structures.yield.from).